persist/fs/commitlog/iterator.go

```diff
@@ -177,3 +171,10 @@ func (i *iterator) nextReader() bool {
 	i.reader = reader
 	return true
 }
+
+func (i *iterator) closeAndResetReader() {
```

Comment:
`i.reader.Close()` can return an error, could you bubble that up here

Reply:
yeah I can return it from this function, but it looked like the code was intentionally swallowing the errors before. Should I make this return the error and have the caller continue to ignore it for now?

Reply:
Yeah we shouldn't drop these errors silently, could you rewire w/e is necessary.